File: VB/GridColumnHeight.Module/Invoice.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.Xpo
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl

Namespace GridColumnHeight.Module
	<NavigationItem()> _
	Public Class Invoice
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)

		End Sub

		#Region "DocumentNumber"

		Private _documentNumber As String

		<Size(10), Indexed(), Index(0)> _
		Public Property DocumentNumber() As String
			Get
				Return _documentNumber
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("DocumentNumber", _documentNumber, value)
			End Set
		End Property

		#End Region

		#Region "DocumentChaseDate"

		Private _documentChaseDate As Nullable(Of DateTime)

		Public Property DocumentChaseDate() As Nullable(Of DateTime)
			Get
				Return _documentChaseDate
			End Get
			Set(ByVal value As Nullable(Of DateTime))
				SetPropertyValue(Of Nullable(Of DateTime))("DocumentChaseDate", _documentChaseDate, value)
			End Set
		End Property

		#End Region

		#Region "Lines"

		<Association()> _
		Public ReadOnly Property Lines() As XPCollection(Of InvoiceLine)
			Get
				Return GetCollection(Of InvoiceLine)("Lines")
			End Get
		End Property

		#End Region

		#Region "RelatedInvoice"

		Private _relatedInvoice As Invoice

		'    [DisplayName("RelatedInvoice")]
		Public Property RelatedInvoice() As Invoice
			Get
				Return _relatedInvoice
			End Get
			Set(ByVal value As Invoice)
				SetPropertyValue(Of Invoice)("RelatedInvoice", _relatedInvoice, value)
			End Set
		End Property

		#End Region

		#Region "SpaceWaster01"

		Private _spaceWaster01 As String

		Public Property SpaceWaster01() As String
			Get
				Return _spaceWaster01
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("SpaceWaster01", _spaceWaster01, value)
			End Set
		End Property

		#End Region

		#Region "SpaceWaster02"

		Private _spaceWaster02 As String

		Public Property SpaceWaster02() As String
			Get
				Return _spaceWaster02
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("SpaceWaster02", _spaceWaster02, value)
			End Set
		End Property

		#End Region

		#Region "SpaceWaster03"

		Private _spaceWaster03 As String

		Public Property SpaceWaster03() As String
			Get
				Return _spaceWaster03
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("SpaceWaster03", _spaceWaster03, value)
			End Set
		End Property

		#End Region

		#Region "SpaceWaster04"

		Private _spaceWaster04 As String

		Public Property SpaceWaster04() As String
			Get
				Return _spaceWaster04
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("SpaceWaster04", _spaceWaster04, value)
			End Set
		End Property

		#End Region

		#Region "SpaceWaster05"

		Private _spaceWaster05 As String

		Public Property SpaceWaster05() As String
			Get
				Return _spaceWaster05
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("SpaceWaster05", _spaceWaster05, value)
			End Set
		End Property

		#End Region

		#Region "SpaceWaster06"

		Private _spaceWaster06 As String

		Public Property SpaceWaster06() As String
			Get
				Return _spaceWaster06
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("SpaceWaster06", _spaceWaster06, value)
			End Set
		End Property

		#End Region

		#Region "SpaceWaster07"

		Private _spaceWaster07 As String

		Public Property SpaceWaster07() As String
			Get
				Return _spaceWaster07
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("SpaceWaster07", _spaceWaster07, value)
			End Set
		End Property

		#End Region

		#Region "SpaceWaster08"

		Private _spaceWaster08 As String

		Public Property SpaceWaster08() As String
			Get
				Return _spaceWaster08
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("SpaceWaster08", _spaceWaster08, value)
			End Set
		End Property

		#End Region

		#Region "SpaceWaster09"

		Private _spaceWaster09 As String

		Public Property SpaceWaster09() As String
			Get
				Return _spaceWaster09
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("SpaceWaster09", _spaceWaster09, value)
			End Set
		End Property

		#End Region

		#Region "SpaceWaster10"

		Private _spaceWaster10 As String

		Public Property SpaceWaster10() As String
			Get
				Return _spaceWaster10
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("SpaceWaster10", _spaceWaster10, value)
			End Set
		End Property

		#End Region

	End Class
End Namespace

File: CS/GridColumnHeight.Module/Invoice.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.Xpo;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;

namespace GridColumnHeight.Module
{
    [NavigationItem()]
    public class Invoice : BaseObject
    {
        public Invoice(Session session)
            : base(session)
        {

        }

        #region DocumentNumber

        private string _documentNumber;

        [Size(10), Indexed(), Index(0)]
        public string DocumentNumber
        {
            get
            {
                return _documentNumber;
            }
            set
            {
                SetPropertyValue<string>("DocumentNumber", ref _documentNumber, value);
            }
        }

        #endregion

        #region DocumentChaseDate

        private DateTime? _documentChaseDate;

        public DateTime? DocumentChaseDate
        {
            get
            {
                return _documentChaseDate;
            }
            set
            {
                SetPropertyValue<DateTime?>("DocumentChaseDate", ref _documentChaseDate, value);
            }
        }

        #endregion

        #region Lines

        [Association()]
        public XPCollection<InvoiceLine> Lines
        {
            get { return GetCollection<InvoiceLine>("Lines"); }
        }
        
        #endregion

        #region RelatedInvoice

        private Invoice _relatedInvoice;

        //    [DisplayName("RelatedInvoice")]
        public Invoice RelatedInvoice
        {
            get
            {
                return _relatedInvoice;
            }
            set
            {
                SetPropertyValue<Invoice>("RelatedInvoice", ref _relatedInvoice, value);
            }
        }

        #endregion

        #region SpaceWaster01

        private string _spaceWaster01;

        public string SpaceWaster01
        {
            get
            {
                return _spaceWaster01;
            }
            set
            {
                SetPropertyValue<string>("SpaceWaster01", ref _spaceWaster01, value);
            }
        }

        #endregion

        #region SpaceWaster02

        private string _spaceWaster02;

        public string SpaceWaster02
        {
            get
            {
                return _spaceWaster02;
            }
            set
            {
                SetPropertyValue<string>("SpaceWaster02", ref _spaceWaster02, value);
            }
        }

        #endregion

        #region SpaceWaster03

        private string _spaceWaster03;

        public string SpaceWaster03
        {
            get
            {
                return _spaceWaster03;
            }
            set
            {
                SetPropertyValue<string>("SpaceWaster03", ref _spaceWaster03, value);
            }
        }

        #endregion
        
        #region SpaceWaster04

        private string _spaceWaster04;

        public string SpaceWaster04
        {
            get
            {
                return _spaceWaster04;
            }
            set
            {
                SetPropertyValue<string>("SpaceWaster04", ref _spaceWaster04, value);
            }
        }

        #endregion
        
        #region SpaceWaster05

        private string _spaceWaster05;

        public string SpaceWaster05
        {
            get
            {
                return _spaceWaster05;
            }
            set
            {
                SetPropertyValue<string>("SpaceWaster05", ref _spaceWaster05, value);
            }
        }

        #endregion
        
        #region SpaceWaster06

        private string _spaceWaster06;

        public string SpaceWaster06
        {
            get
            {
                return _spaceWaster06;
            }
            set
            {
                SetPropertyValue<string>("SpaceWaster06", ref _spaceWaster06, value);
            }
        }

        #endregion
        
        #region SpaceWaster07

        private string _spaceWaster07;

        public string SpaceWaster07
        {
            get
            {
                return _spaceWaster07;
            }
            set
            {
                SetPropertyValue<string>("SpaceWaster07", ref _spaceWaster07, value);
            }
        }

        #endregion
        
        #region SpaceWaster08

        private string _spaceWaster08;

        public string SpaceWaster08
        {
            get
            {
                return _spaceWaster08;
            }
            set
            {
                SetPropertyValue<string>("SpaceWaster08", ref _spaceWaster08, value);
            }
        }

        #endregion
        
        #region SpaceWaster09

        private string _spaceWaster09;

        public string SpaceWaster09
        {
            get
            {
                return _spaceWaster09;
            }
            set
            {
                SetPropertyValue<string>("SpaceWaster09", ref _spaceWaster09, value);
            }
        }

        #endregion
        
        #region SpaceWaster10

        private string _spaceWaster10;

        public string SpaceWaster10
        {
            get
            {
                return _spaceWaster10;
            }
            set
            {
                SetPropertyValue<string>("SpaceWaster10", ref _spaceWaster10, value);
            }
        }

        #endregion
        
    }
}
